src/scanoss/file_filters.py

```diff
@@ -252,57 +260,85 @@ def __init__(
             all_extensions (bool): Include all file extensions
             all_folders (bool): Include all folders
             hidden_files_folders (bool): Include hidden files and folders
+            operation_type: operation type. can be either 'scanning' or 'fingerprinting'
         """
         super().__init__(debug, trace, quiet)
 
+        if skip_folders is None:
+            skip_folders = []
+        if skip_extensions is None:
+            skip_extensions = []
         self.hidden_files_folders = hidden_files_folders
         self.scanoss_settings = scanoss_settings
         self.all_extensions = all_extensions
         self.all_folders = all_folders
+        self.skip_folders = skip_folders
+        self.skip_size = skip_size
+        self.skip_extensions = skip_extensions
+        self.file_folder_pat_spec = self._get_file_folder_pattern_spec(operation_type)
+        self.size_pat_rules = self._get_size_limit_pattern_rules(operation_type)
 
-    def get_filtered_files_from_folder(self, root: str, operation_type: str) -> List[str]:
-        """Retrieve a list of files to scan or fingerprint from a given directory root based on filter settings.
+    def get_filtered_files_from_folder(self, root: str) -> List[str]:
+        """
+        Retrieve a list of files to scan or fingerprint from a given directory root based on filter settings.
 
         Args:
             root (str): Root directory to scan or fingerprint
-            operation_type (str): Type of operation ('scanning' or 'fingerprinting')
 
         Returns:
             list[str]: Filtered list of files to scan or fingerprint
         """
+        if self.debug:
```

Comment:
here we could remove this if check 

you can just do `self.print_debug` and it will do it